### PR TITLE
Add Azure artifact archival functions, for use with the Azure Storage Jenkins plugin.

### PR DESCRIPTION
**Pull request by crummel**
This mostly just copies the existing archival functions and their options.

### CODE REVIEW

**Comment by crummel** [comment]
cc @mmitche

File: jobs/generation/Utilities.groovy [hunk 1]
```diff
@@ -837,6 +872,38 @@ class Utilities {
 
         addArchival(job, settings)
     }
+
```
Comment (mmitche) [inline jobs/generation/Utilities.groovy]: You can just eliminate the deprecated version.  We want to move to the ArchivalSettings version when possible.

File: jobs/generation/Utilities.groovy [hunk 2]
```diff
@@ -806,6 +806,41 @@ class Utilities {
...
+                                uploadArtifactsOnlyIfSuccessful(false)
+                                manageArtifacts(true)
+                                uploadZips(false)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (mmitche) [inline jobs/generation/Utilities.groovy]: This part looks good.

File: jobs/generation/Utilities.groovy (hunk 1 above)
Comment (crummel) [inline jobs/generation/Utilities.groovy]: Removed this.

**Pull request merged**